sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -213,12 +217,30 @@ case class FilterExec(condition: Expression, child: SparkPlan)
   protected override def doExecute(): RDD[InternalRow] = {
     val numOutputRows = longMetric("numOutputRows")
     child.execute().mapPartitionsWithIndexInternal { (index, iter) =>
-      val predicate = newPredicate(condition, child.output)
-      predicate.initialize(0)
-      iter.filter { row =>
-        val r = predicate.eval(row)
-        if (r) numOutputRows += 1
-        r
+      try {
+        val predicate = newPredicate(condition, child.output)
+        predicate.initialize(0)
+        iter.filter { row =>
+          val r = predicate.eval(row)
+          if (r) numOutputRows += 1
+          r
+        }
+      } catch {
+        // JaninoRuntimeException is in a nested exception if Java compilation error occurs
+        case e: Exception if ExceptionUtils.getRootCause(e).isInstanceOf[JaninoRuntimeException] =>
```

Comment:
Rather than walk the exception tree, should we just make [our wrapping](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala#L957) more specific?